_sources/Debugging/intro-HowtobeaSuccessfulProgrammer.rst

```diff
@@ -7,10 +7,230 @@
+
+What is important?  The most obvious thing you will need at a higher level of recall is what
+*just messed you up*, what you missed until doing this problem:  After finishing the
+actual problem, *actively* follow up and ask yourself:
+
+- What did I get in the end that I was missing initially? What was the connection I made?
+- Does this example fit in to some larger idea/abstraction/generalization in a way that
+  I did not see before?
+- How am I going to look at this so I can make a similar connection
+  in a similar (or maybe only partly similar) problem?
+- Is there a kernel here that I can think of as a new tool in my bag of tricks?
+  
+It is hard to put in this pause for reflection! 
+However your answers to these questions are the most important things to take away from your
+recent hard work.  
+The extra consideration puts them more in
+the "priority" part of your brain, so you can really learn from your effort.  When you need 
+the important ideas 
+next, you do not need to play through all the details of 
+the stuff you did to solve the exact earlier problem.
+
+For example, if you had trouble with the exercise mentioned above about days of the week,
+hopefully you ended up concentrating on a takeaway something like: 
+"I can deal with things that are cyclic."  
+(Then, surprise! Several of the other exercises in that set get a lot easier.)
+
+Keep coming back to the processes in this section:  
+It is really important as you have more and more to juggle.
 
-One of the most important skills you need to aquire to complete this book successfully is the ability to debug your programs.  Debugging might be the most under-appreciated, and under-taught, skill in introductory computer science.  For that reason we are introducing a series of "debugging interludes."  Debugging is a skill that you need to master over time, and some of the tips and tricks are specific to different aspects of Python programming.  So look for additional debugging interludes throughout the rest of this book.
 
-Programming is an odd thing in a way.  Here is why.  As programmers we spend 99% of our time trying to get our program to work.  We struggle, we stress, we spend hours deep in frustration trying to get our program to execute correctly.  Then when we do get it going we celebrate, hand it in, and move on to the next homework assignment or programming task.  But here is the secret, when you are successful, you are happy, your brain releases a bit of chemical that makes you feel good.  You need to organize your programming so that you have lots of little successess.  It turns out your brain doesn't care all that much if you have successfully written hello world, or a fast fourier transform (trust me its hard) you still get that little release that makes you happy.  When you are happy you want to go on and solve the next little problem.  Essentially I'm telling you once again, start small, get something small working, and then add to it.
```

Comment:
I like this paragraph.  I would rather keep it.

Reply:
This was originally in 3.1.  Part of this is repeated.  3.2 has

Every time you have a little success, your brain releases a tiny bit of chemical that makes you happy. So, you can keep yourself happy and make programming more enjoyable by creating lots of small victories for yourself.

In latest push I replaced this duplicated part by your whole paragraph, now in 3.2.

Reply:
I think pushing it was enough to get it added to this still unaccepted pull request.